src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const newDateArray = [];
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = dateArray[fromFormat.indexOf('YY')];
+
+    dateArray[fromFormat.indexOf('YY')] = year < 30 ? '20' + year : '19' + year;
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    const year = dateArray[fromFormat.indexOf('YYYY')];
+
+    dateArray[fromFormat.indexOf('YYYY')] = year[2] + year[3];
+  }
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
```

Comment:
The nested loop starting here is inefficient because it unnecessarily iterates over all combinations of `fromFormat` and `toFormat` elements. You can achieve the same result with a single loop by directly accessing the indices of the date parts in the `toFormat` array.